shopinvader_product_order_mode/models/product_product.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2025 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    shop_order_mode = fields.Selection(
+        selection=[("direct_sale_only", _("Direct Sale Only"))],
+        string="Shopinvader Order Mode",
+        store=True,
+        readonly=False,
+        compute="_compute_shop_order_mode",
+        inverse="_inverse_shop_order_mode",
+        help="Order mode for the product in Shopinvader.",
+    )
+    is_shop_order_mode_unabled = fields.Boolean(
+        string="Template Shop Order Mode Disabled",
+        related="product_tmpl_id.is_shop_order_mode_unabled_on_variant",
+        store=True,
+        readonly=True,
+    )
+
+    @api.depends("product_tmpl_id.shop_order_mode")
+    def _compute_shop_order_mode(self):
+        for product in self:
+            if product.product_tmpl_id:
+                product.shop_order_mode = product.product_tmpl_id.shop_order_mode
+            else:
+                product.shop_order_mode = False
```

Comment:
With this you should not need to hack the write anymore

```suggestion
    @api.depends(
         "product_tmpl_id.shop_order_mode",
         "product_tmpl_id.is_shop_order_mode_unabled_on_variant"
         )
    def _compute_shop_order_mode(self):
        for product in self:
            if product.product_tmpl_id:
                if product_tmpl_id.is_shop_order_mode_unabled_on_variant:
                    product.shop_order_mode = product.product_tmpl_id.shop_order_mode
            else:
                product.shop_order_mode = False
```